server/channels/api4/user.go

```diff
@@ -3445,3 +3446,59 @@ func getUsersWithInvalidEmails(c *Context, w http.ResponseWriter, r *http.Reques
 		c.Logger.Warn("Error writing response", mlog.Err(err))
 	}
 }
+
+func resetUserAuthDataToEmail(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSystem) {
+		c.SetPermissionError(model.PermissionManageSystem)
+		return
+	}
+	type ResetAuthDataParams struct {
+		AuthService      string   `json:"auth_service"`
+		IncludeDeleted   bool     `json:"include_deleted"`
+		DryRun           bool     `json:"dry_run"`
+		SpecifiedUserIDs []string `json:"user_ids"`
+	}
+	var params *ResetAuthDataParams
+	jsonErr := json.NewDecoder(r.Body).Decode(&params)
+	if jsonErr != nil || params == nil {
+		c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "model.utils.decode_json.app_error", nil, "", http.StatusBadRequest).Wrap(jsonErr)
+		return
+	}
+
+	authService := params.AuthService
+
+	// Check the auth service matches our offerings
+	if authService == "" || (authService != model.UserAuthServiceSaml && authService != model.ServiceOpenid && authService != model.UserAuthServiceGitlab && authService != model.ServiceGoogle && authService != model.ServiceOffice365) {
+		c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "api.admin.users.invalid_auth_type.app_error", map[string]any{"AuthService": authService}, "", http.StatusBadRequest)
+		return
+	}
+
+	licenseErr := minimumProfessionalLicense(c)
+
+	// Gitlab oauth is part of our free offering but gitlab openId requires a professional license.
+	if authService == model.UserAuthServiceGitlab && c.App.Config().IsSSOServiceUsingOpenId(authService) && licenseErr != nil {
+		c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "api.admin.users.invalid_auth_type.gitlab_license.app_error", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	if (authService == model.ServiceOpenid || authService == model.ServiceGoogle || authService == model.ServiceOffice365 || authService == model.UserAuthServiceSaml) && licenseErr != nil {
+		c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "api.admin.users.invalid_auth_type.license_level.app_error", map[string]any{"AuthService": authService}, "", http.StatusNotImplemented)
+		return
+	}
```

Comment:
The check seems overly complicated. I wonder if it's easier to read with a `switch` statement over `authService`.

Something like
```golang
switch authService {
	case model.UserAuthServiceGitlab:
		if c.App.Config().IsSSOServiceUsingOpenId(authService) && licenseErr != nil {
			c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "api.admin.users.invalid_auth_type.gitlab_license.app_error", nil, "", http.StatusNotImplemented)
			return
		}
	default:
		if licenseErr != nil {
			c.Err = model.NewAppError("api.resetUserAuthDataToEmail", "api.admin.users.invalid_auth_type.license_level.app_error", map[string]any{"AuthService": authService}, "", http.StatusNotImplemented)
			return
		}	
}
```